xdem/coreg/biascorr.py

```diff
@@ -176,17 +179,29 @@ def _fit_biascorr(  # type: ignore
         # Compute difference and mask of valid data
         # TODO: Move the check up to Coreg.fit()?
 
-        diff = ref_elev - tba_elev
-        valid_mask = np.logical_and.reduce(
-            (inlier_mask, np.isfinite(diff), *(np.isfinite(var) for var in bias_vars.values()))
-        )
+        if all(isinstance(dem, (da.Array, Delayed)) for dem in (ref_elev, tba_elev)):
+            diff = da.subtract(ref_elev, tba_elev)
+
```

Comment:
We also need to compute the `valid_mask` here for the delayed inputs to ensure we don't introduce outliers from the `ref_elev` or `tba_elev`, same as below:
```python
            valid_mask = np.logical_and.reduce(
                (inlier_mask, np.isfinite(diff), *(np.isfinite(var) for var in bias_vars.values()))
            )
```
Not sure if this is supported directly, or if we need to use `map_blocks` for it...

Reply:
I guess `map_blocks` the same way you used it further up would work perfectly :)

Reply:
So the reason I have not calculated this is because I calculate the inlier_mask from all of the inputs [here](https://github.com/GlacioHack/xdem/pull/525/files#diff-1430c7a606ab459441322127ec6b1c671169f487050d53481cc23723099bc834R344) . Taking also the non-valid values into account from the `ref_elev` and `tba_elev`. I think this would cover that? Or did I miss something?

Reply:
It's for the `bias_vars` (that can have different non-finite values that will propagate during the fit), not covered by ` _get_valid_data_mask`! 
And actually this will also be relevant in affine functions (for instance for slope/aspect in `NuthKaab` that propagate NaNs of the DEM)

Reply:
ahhh ok ok. I missed that thank you! :+1: ill make sure to add it :slightly_smiling_face:

Reply:
this shouldn't have an effect on the `Deramp` algorithm though since the `bias_vars` is created from the input coordinates of the `reference_elevation` input correct? I'm just asking because I'm running into some unexpected outputs when I have nans in the original input data.

Reply:
Yes it shouldn't have any effect in this case, the meshgrid should be always a full array :thinking:

Reply:
Ok yes that makes sense. I did end up figuring out what the issue was with the nan data. I was calculating the mask on the output from the `delayed_reproject` ([here](https://github.com/GlacioHack/xdem/pull/525/commits/32eb94b31acafa38855f6dcf8fefe558ecb4e6a9) is the change). I'm not 100% sure, but my assumption is that there is some smoothing/ resampling happening and i guess the nodata values accidentally get smeared into neighboring pixels?... Meaning it was potentially sampling some very high values and the estimated function was pretty off. This was my assumption after looking at the input y values going into the fit function.